lib/service/service.go

```diff
@@ -3966,41 +3972,24 @@ func (process *TeleportProcess) initProxyEndpoint(conn *Connector) error {
 		}
 	}
 
-	var grpcServer *grpc.Server
+	var (
+		grpcServerInsecure *grpc.Server
+		grpcServerSecure   *grpc.Server
+	)
 	if alpnRouter != nil {
-		grpcServer = grpc.NewServer(
-			grpc.ChainUnaryInterceptor(
-				utils.GRPCServerUnaryErrorInterceptor,
-				proxyLimiter.UnaryServerInterceptor(),
-			),
-			grpc.ChainStreamInterceptor(
-				utils.GRPCServerStreamErrorInterceptor,
-				proxyLimiter.StreamServerInterceptor,
-			),
-			grpc.KeepaliveParams(keepalive.ServerParameters{
-				// Using an aggressive idle timeout here since this gRPC server
-				// currently only hosts the join service, which has no need for
-				// long-lived idle connections.
-				//
-				// The reason for introducing this is that teleport clients
-				// before #17685 is fixed will hold connections open
-				// indefinitely if they encounter an error during the joining
-				// process, and this seems like the best way for the server to
-				// forcibly close those connections.
-				//
-				// If another gRPC service is added here in the future, it
-				// should be alright to increase or remove this idle timeout as
-				// necessary once the client fix has been released and widely
-				// available for some time.
-				MaxConnectionIdle: 10 * time.Second,
-			}),
+		grpcServerInsecure = process.initInsecureGRPCServer(log, proxyLimiter, conn, listeners.grpcInsecure)
+		grpcTLSConfig, err := conn.ServerIdentity.TLSConfig(cfg.CipherSuites)
```

Comment:
Since `conn` is required  as a initSecureGRPCServer param can we consider moving this call to the `initSecureGRPCServer`  function ?